config/clients/go/template/client/client.mustache

```diff
@@ -2077,6 +2120,264 @@ func (client *{{appShortName}}Client) BatchCheckExecute(request SdkClientBatchCh
+	ctx := request.GetContext()
+	body := request.GetBody()
+	options := request.GetOptions()
+
+	if body == nil || len(body.Checks) == 0 {
+		return nil, FgaRequiredParamError{param: "checks"}
+	}
+
+	if options == nil {
+		options = &BatchCheckOptions{}
+	}
+
+	maxParallelRequests := int32(10)
+	if options.MaxParallelRequests != nil {
+		maxParallelRequests = *options.MaxParallelRequests
+	}
+
+	maxBatchSize := int32(50)
+	if options.MaxBatchSize != nil {
+		maxBatchSize = *options.MaxBatchSize
+	}
+
+	_, err := client.getStoreId(options.StoreId)
+	if err != nil {
+		return nil, err
+	}
+
+	authorizationModelId, err := client.getAuthorizationModelId(options.AuthorizationModelId)
+	if err != nil {
+		return nil, err
+	}
+
+	chunks := chunkClientBatchCheckItems(body.Checks, int(maxBatchSize))
+
+	resultChan := make(chan *fgaSdk.BatchCheckResponse)
+	errorChan := make(chan error)
+
+	var wg errgroup.Group
+	wg.SetLimit(int(maxParallelRequests))
+
+	for _, chunk := range chunks {
+		chunkCopy := chunk
+
+		wg.Go(func() error {
+			batchCheckRequest := createBatchCheckRequest(chunkCopy, authorizationModelId, options.Consistency)
+
+			response, err := client.SingleBatchCheck(ctx, batchCheckRequest, options)
+			if err != nil {
+				errorChan <- err
+				return err
+			}
+
+			resultChan <- response
+			return nil
+		})
+	}
+
+	go func() {
+		err := wg.Wait()
+		if err != nil {
+			errorChan <- err
+		}
+		close(resultChan)
+		close(errorChan)
+	}()
+
+	var results []*fgaSdk.BatchCheckResponse
+	for response := range resultChan {
+		results = append(results, response)
+	}
+
+	select {
+	case err := <-errorChan:
+		if err != nil {
+			return nil, err
+		}
+	default:
+		// No error
+	}
+
+	combinedResult := make(map[string]fgaSdk.BatchCheckSingleResult)
+
+	for _, response := range results {
+		for correlationID, result := range response.GetResult() {
+			combinedResult[correlationID] = result
+		}
+	}
+
+	combinedResponse := fgaSdk.NewBatchCheckResponse()
+	combinedResponse.SetResult(combinedResult)
+
+	return combinedResponse, nil
+}
+
+/*
+ * SingleBatchCheck performs a single batch check request to the API
+ * @param ctx _context.Context - for authentication, logging, cancellation, deadlines, tracing, etc.
+ * @param body fgaSdk.BatchCheckRequest - the request body
+ * @param options *BatchCheckOptions - options for the request
+ * @return *fgaSdk.BatchCheckResponse
+ */
+func (client *OpenFgaClient) SingleBatchCheck(ctx _context.Context, body fgaSdk.BatchCheckRequest, options *BatchCheckOptions) (*fgaSdk.BatchCheckResponse, error) {
+	storeId, err := client.getStoreId(options.StoreId)
+	if err != nil {
+		return nil, err
+	}
+
+	req := client.OpenFgaApi.BatchCheck(ctx, *storeId)
+	req = req.Body(body)
+
+	requestID := fmt.Sprintf("%d", time.Now().UnixNano())
+	client.APIClient.GetConfig().AddDefaultHeader("X-Fga-Client-Bulk-Request-Id", requestID)
```

Comment:
This will set the header on all requests, we can't do this here

Reply:
Resolved in [23ee897](https://github.com/openfga/sdk-generator/pull/511/commits/23ee897d3cbde548c67932fc3bc557c3fe8724d3)